python_modules/dagster/dagster/_core/execution/plan/state.py

```diff
@@ -29,6 +31,27 @@
     from dagster._core.execution.plan.plan import StepHandleUnion
 
 
+@whitelist_for_serdes
+class AssetProvenanceData(NamedTuple):
+    asset_key: AssetKey
+    logical_version_provenance: LogicalVersionProvenance
+
+    @staticmethod
+    def list_from_map(
```

Comment:
Given that each of these static methods is just used in a single place and the functionality is fairly generic collections manipulation, could the implementations just be inlined?

Reply:
I would prefer to leave it as is because it directly mirrors the structure and handling of`StepOutputVersionData`. I would like to eliminate both of these in one stroke later.